src/main/java/nextstep/subway/sections/domain/Sections.java

```diff
@@ -0,0 +1,169 @@
+        }
+    }
+
+    public List<Section> getSections() {
+        return Collections.unmodifiableList(sections);
+    }
+
+    public List<Station> orderedStations() {
+        if (sections.isEmpty()) {
+            return Arrays.asList();
+        }
+
+        List<Station> stations = new ArrayList<>();
+        Station downStation = findUpStation();
+        stations.add(downStation);
+
+        while (downStation != null) {
+            Station finalDownStation = downStation;
+            Section nextSection = findHasSameUpStationSection(finalDownStation);
+            if (nextSection == null) {
+                break;
+            }
+            downStation = nextSection.getDownStation();
+            stations.add(downStation);
+        }
+
+        return Collections.unmodifiableList(stations);
+    }
+
+    private Station findUpStation() {
+        Station downStation = sections.get(FIRST_INDEX).getUpStation();
+        while (downStation != null) {
+            Station finalDownStation = downStation;
+            Section nextSection = findHasSameDownStationSection(finalDownStation);
+            if (nextSection == null) {
+                break;
+            }
+            downStation = nextSection.getUpStation();
+        }
+
+        return downStation;
+    }
+
+    public void delete(Station willDeleteStation) {
+        if (sections.size() <= LOWER_SECTION_SIZE) {
+            throw new RuntimeException("구간이 하나뿐일 때는 삭제할 수 없습니다.");
+        }
+
+        Section hasSameUpStationSection = findHasSameUpStationSection(willDeleteStation);
+        Section hasSameDownStationSection = findHasSameDownStationSection(willDeleteStation);
+
+        if (hasSameUpStationSection != null && hasSameDownStationSection != null) {
+            makeConnectSection(hasSameUpStationSection, hasSameDownStationSection);
+        }
+
+        removeSection(hasSameUpStationSection, hasSameDownStationSection);
+    }
+
+    private void makeConnectSection(Section hasUpStationSection, Section hasDownStationSection) {
+        Station newUpStation = hasDownStationSection.getUpStation();
+        Station newDownStation = hasUpStationSection.getDownStation();
+        int newDistance =
+            hasUpStationSection.getDistance() + hasDownStationSection.getDistance();
+        sections.add(new Section(hasUpStationSection.getLine(), newUpStation, newDownStation,
+            newDistance));
+    }
+
+    private void removeSection(Section hasUpStationSection, Section hasDownStationSection) {
+        if (hasUpStationSection != null) {
+            sections.remove(hasUpStationSection);
+        }
+        if (hasDownStationSection != null) {
+            sections.remove(hasDownStationSection);
+        }
+    }
+
+
+    private Section findHasSameUpStationSection(Station station) {
+        return sections.stream().filter(it -> it.getUpStation().equals(station)).findFirst()
+            .orElse(null);
+    }
```

Comment:
개인적으로는 stream 이후로는 개행을 해주는 것이 가독성이 좋다고 생각하는데 어떠신가요? 😄 
```suggestion
    private Section findHasSameUpStationSection(Station station) {
        return sections.stream()
                .filter(it -> it.getUpStation().equals(station))
                .findFirst()
                .orElse(null);
    }
```

Reply:
저도 훨씬 가독성이 좋은거 같아요 😺  수정했습니다!